src/orange/orangeinp/detail/PolygonUtils.hh

```diff
@@ -254,6 +254,23 @@ filter_collinear_points(std::vector<Real2> const& corners, double abs_tol)
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the min/max values of a polygon for a given dimension.
+ */
+inline std::pair<Real2::value_type, Real2::value_type>
+find_extrema(std::vector<Real2> const& polygon, size_type dim)
```

Comment:
I added this back when I was using a different approach. It is used in `ExtrudedPolygon` within this MR. I still think it's a good refactor, even if it's not used in `RevolvedSpecialTrapezoid`

Reply:
This is a carry-over from the original implementation, in which this function would apply to both polygons on the XY plane and RZ plane. Now it's only used in the XY plane, so `Axis` technically works, but it makes it less generic.